workflow-controller/controller-service/src/main/java/io/hyscale/controller/hooks/VolumeValidatorHook.java

```diff
@@ -329,14 +329,16 @@ private void initStorageClass(ApiClient apiClient) throws HyscaleException {
 		}
 	}
 
-	private boolean matchStorageClass(String existing, String modified) {
-		if (StringUtils.isBlank(modified)) {
-			return (existing != null && getDefaultStorageClass() != null) ? existing.equals(getDefaultStorageClass())
-					: false;
-		}
-		if (StringUtils.isNotBlank(existing) && StringUtils.isNotBlank(modified)) {
-			return existing.equals(modified);
-		}
-		return false;
-	}
+    private boolean matchStorageClass(String existing, String modified) {
+        if (StringUtils.isBlank(modified)) {
+            List<String> defaultStorageClassList = getDefaultStorageClass();
+            return (existing != null && defaultStorageClassList != null && defaultStorageClassList.size() == 1)
+                    ? existing.equals(defaultStorageClassList.get(0))
+                    : false;
```

Comment:
Why only with 0 index ?
It can match with any default storage class right ?

Reply:
If user is not providing default storage class, and cluster have more than 1, we will fail even before reaching here.
This only needs to cover the case of single default storage class